### PR TITLE
feat(playground): block navigation when runs are in progress

### LINKED ISSUE #5460: [playground][enhancement] add text to inform the user a playground run is still going even if some dataset examples have completed

### LINKED ISSUE #5464: [playground][ui] streaming toggle should be remembered

### LINKED ISSUE #5466: [playground][ui] update save config button text and icon
Users are confused by the save config button - it saves the config as the provider default, users thing it saves the current set of params for running. Make that more apparent

### PR DESCRIPTION
**Pull request by Parker-Stafford**
resolves #5464 
resolves #5460
resolves #5466 

- remembers streaming setting from playground in preferences store
- adds text to inform users that playground is still running and navigating may result in data los
- updates save config button and icon to be more clear as to its purpose (saving config as default, not applying the config to the current run
- small style tweaks on model config form

### Remember streaming preference

https://github.com/user-attachments/assets/48ef1ea0-cecf-47a0-a703-04588acbd656

### Run in progress warnings
Any feedback on wording here is much appreciated

https://github.com/user-attachments/assets/76f1910f-562a-4836-ad14-f10f747049b1

### Save config button

image

### Model config style tweaks

Screenshot 2024-11-20 at 3 55 06 PM

### CODE REVIEW

File: app/src/pages/playground/ModelConfigButton.tsx [hunk 1]
```diff
@@ -204,12 +223,18 @@ function ModelConfigDialog(props: ModelConfigDialogProps) {
...
+            onClick={onSaveConfig}
+            icon={<Icon svg={<Icons.SaveOutline />} />}
+          >
+            Save as Default
           </Button>
           <Tooltip>
-            Remember configuration for{" "}
-            {ModelProviders[instance.model.provider] ?? "this provider"}.
+            Saves the current configuration as the default for{" "}
+            {ModelProviders[instance.model.provider] ?? "this provider"} for
```
Comment (cephalization) [inline app/src/pages/playground/ModelConfigButton.tsx:236]: The "for later use" kinda makes this message a tongue twister.

Two back to back "for"s feels redundant

File: app/src/pages/playground/ModelConfigButton.tsx [hunk 2]
```diff
@@ -322,29 +347,30 @@ function ModelConfigDialogContent(props: ModelConfigDialogContentProps) {
...
-            />
-          )}
-          <Suspense>
-            <InvocationParametersForm instanceId={playgroundInstanceId} />
-          </Suspense>
-        </Flex>
-      </Form>
-    </View>
+    <form
+      onSubmit={(e) => {
```
Comment (cephalization) [inline app/src/pages/playground/ModelConfigButton.tsx:351]: Can this close the dialog? Would be nice to close on enter

File: app/src/pages/playground/PlaygroundDatasetSection.tsx [hunk 3]
```diff
@@ -47,21 +48,30 @@ export function PlaygroundDatasetSection({ datasetId }: { datasetId: string }) {
...
-                );
-              }}
-            >
-              View Experiment
-            </Button>
-          )}
+          <Flex gap={"size-100"} alignItems={"center"}>
+            {isRunning && (
+              <Text color="text-700">
+                Run in progress, navigating away from the page could result in
```
Comment (cephalization) [inline app/src/pages/playground/PlaygroundDatasetSection.tsx:54]: The second part of this message feels redundant. I'd personally rather let the user discover our lack of job based runs if they try to leave first 😅

**Review by cephalization — APPROVED** [review]

File: app/src/pages/playground/ModelConfigButton.tsx (hunk 1 above)
Comment (Parker-Stafford) [reply app/src/pages/playground/ModelConfigButton.tsx:236]: yeah i agree maybe will just leave it in th etoast but remove here

File: app/src/pages/playground/ModelConfigButton.tsx (hunk 2 above)
Comment (Parker-Stafford) [reply app/src/pages/playground/ModelConfigButton.tsx:351]: good call

File: app/src/pages/playground/PlaygroundDatasetSection.tsx (hunk 3 above)
Comment (Parker-Stafford) [reply app/src/pages/playground/PlaygroundDatasetSection.tsx:54]: Yeah i think that makes ense, the loader on the button probably should alert them to that so maybe this text is unnecessary. But i'll leave Run in progress for now

File: app/src/pages/playground/ModelConfigButton.tsx (hunk 2 above)
Comment (Parker-Stafford) [reply app/src/pages/playground/ModelConfigButton.tsx:351]: looks like the form fields arent' triggering submission anyways going to just remove and punt

**Pull request merged**